pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -931,6 +972,11 @@ public void goOnlineFromConsuming(RealtimeSegmentZKMetadata metadata)
   }
 
   protected void downloadSegmentAndReplace(LLCRealtimeSegmentZKMetadata metadata) {
+    closePartitionLevelConsumer();
```

Comment:
similar close and check is needed in buildSegmentFormCommit method?

Reply:
👍 Put the close method in buildSegmentInternal to uniform the behavior